server/src/main/scala/org/scassandra/server/priming/cors/CorsSupport.scala

```diff
@@ -0,0 +1,43 @@
+package org.scassandra.server.priming.cors
+
+import com.typesafe.scalalogging.LazyLogging
+import org.scassandra.server.ScassandraConfig
+import spray.http.HttpHeaders._
+import spray.http.HttpMethods._
+import spray.http.{AllOrigins, HttpMethods, HttpResponse}
+import spray.routing._
+
+/**
+  * A mixin to provide support for providing CORS headers as appropriate
+  */
+trait CorsSupport {
+  this: HttpService with LazyLogging =>
+
+  private val allowOriginHeader = `Access-Control-Allow-Origin`(AllOrigins)
```

Comment:
Yes indeed, adding ‘*’ = AllOrigins is to permissive. I’ve a good solution for this, will do it on the weekend.

Basically my idea is to have a cors-origins list on the config, and when you perform a request from a given domain X, if X is in the list, then we send a `Access-Control-Allow-Origin`(X).
Without sending all the allowed origins back to the client.


> On 31 May 2016, at 13:11, Andrew Tolbert <notifications@github.com> wrote:
> 
> In server/src/main/scala/org/scassandra/server/priming/cors/CorsSupport.scala <https://github.com/scassandra/scassandra-server/pull/162#discussion_r65169556>:
> 
> > +package org.scassandra.server.priming.cors
> > +
> > +import com.typesafe.scalalogging.LazyLogging
> > +import org.scassandra.server.ScassandraConfig
> > +import spray.http.HttpHeaders._
> > +import spray.http.HttpMethods._
> > +import spray.http.{AllOrigins, HttpMethods, HttpResponse}
> > +import spray.routing._
> > +
> > +/**
> > +  * A mixin to provide support for providing CORS headers as appropriate
> > +  */
> > +trait CorsSupport {
> > +  this: HttpService with LazyLogging =>
> > +
> > +  private val allowOriginHeader = `Access-Control-Allow-Origin`(AllOrigins)
> It would be a nice eventual enhancement to be able to restrict the origins instead of allowing everything. That would require adding another config option to ScassandraConfig. Don't think that is needed immediately, but would be nice to eventually have.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/scassandra/scassandra-server/pull/162/files/25f99748f09f8a6290c5b72e805bde6764026c8e#r65169556>, or mute the thread <https://github.com/notifications/unsubscribe/AEcH55u3teoxvRTnKqw54JEtMYbo9MCaks5qHCWJgaJpZM4Io98s>.
>